src/lseq/ident.rs

```diff
@@ -73,7 +59,7 @@ impl<A: Actor> IdentGen<A> {
     /// The absolute largest possible node.
     pub fn upper(&self) -> Identifier<A> {
         Identifier {
-            path: vec![(2u64.pow(self.initial_base_bits) - 1, None)],
+            path: vec![(self.width_at(0), None)],
```

Comment:
note: I think that ` - 1` was a bug. `2u64.pow(self.initial_base_bits) - 1`  is a valid index but `None < Some(_)`. So it's possible to construct a valid identifier larger than `IdentGen::upper()`.

Changed it to just use the max width for depth 0 (`2u64.pow(self.initial_base_bits)`)